### PR TITLE
textscreen zenity chooser file filter tweaks

### PR DESCRIPTION
**Pull request by jmtd**
Two tweaks

 * adjust patterns to be case-insensitive
 * add a *.* pattern so user can select any file if they wish

### CODE REVIEW

File: textscreen/txt_fileselect.c [hunk 1]
```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
     return ZenityAvailable();
 }
 
+/*
+ * ExpandExtension
+ * given an extension (like wad)
+ * return a pointer to a string that is a case-insensitive
+ * pattern representation (like [Ww][Aa][Dd])
+ */
+static char * ExpandExtension(char *orig)
```
Comment (fragglet) [inline textscreen/txt_fileselect.c]: Remove space

File: textscreen/txt_fileselect.c [hunk 2]
```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
     return ZenityAvailable();
 }
 
+/*
```
Comment (fragglet) [inline textscreen/txt_fileselect.c]: /* */ comments are unusual. Please just use // for consistency.

File: textscreen/txt_fileselect.c [hunk 3]
```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
...
+ * pattern representation (like [Ww][Aa][Dd])
+ */
+static char * ExpandExtension(char *orig)
+{
+    int oldlen,newlen, i;
+    char *c,*newext = NULL;
+
+    oldlen = strlen(orig);
+    newlen = oldlen * 4; // pathalogical case: 'w' => '[Ww]'
+    newext = (char *)malloc(newlen+1);
```
Comment (fragglet) [inline textscreen/txt_fileselect.c]: There's no need to cast void * pointers in C.

File: textscreen/txt_fileselect.c [hunk 4]
```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
...
+    if (newext)
```
Comment (fragglet) [inline textscreen/txt_fileselect.c]: Better to check for the error case, eg.

    if (newext == NULL)
    {
        return NULL;
    }

File: textscreen/txt_fileselect.c [hunk 5]
```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
...
+                *c++ = tolower(orig[i]);
+                *c++ = toupper(orig[i]);
+                *c++ = ']';
+            }
+            else
+            {
+                *c++ = orig[i];
+            }
+        }
+        *c = 0;
```
Comment (fragglet) [inline textscreen/txt_fileselect.c]: Kind of a nitpick but I usually use '\0' to indicate that it's a character.

File: textscreen/txt_fileselect.c [hunk 6]
```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
...
+ * return a pointer to a string that is a case-insensitive
+ * pattern representation (like [Ww][Aa][Dd])
+ */
+static char * ExpandExtension(char *orig)
+{
+    int oldlen,newlen, i;
+    char *c,*newext = NULL;
+
+    oldlen = strlen(orig);
+    newlen = oldlen * 4; // pathalogical case: 'w' => '[Ww]'
```
Comment (fragglet) [inline textscreen/txt_fileselect.c]: pathological :)

Comment (fragglet) [inline textscreen/txt_fileselect.c]: Don't you need oldlen * 4 + 1 to include the \0?

Comment (jmtd) [inline textscreen/txt_fileselect.c]: Yes, and I +1 in the malloc itself; but it could be done here instead.

**Comment by fragglet** [comment]
Thanks!

**Pull request merged**